tools/pd-simulator/simulator/client.go

```diff
@@ -263,13 +269,16 @@ func (c *client) Bootstrap(ctx context.Context, store *metapb.Store, region *met
 	if err != nil {
 		return err
 	}
-	_, err = c.pdClient().Bootstrap(ctx, &pdpb.BootstrapRequest{
+	res, err := c.pdClient().Bootstrap(ctx, &pdpb.BootstrapRequest{
 		Header: c.requestHeader(),
 		Store:  proto.Clone(store).(*metapb.Store),
 		Region: proto.Clone(region).(*metapb.Region),
 	})
-	if err != nil {
-		return err
+	if err != nil || res.GetHeader().GetError() != nil {
+		if err != nil {
+			return err
+		}
+		return errors.Errorf("bootstrap failed: %s", resp.Header.GetError().String())
 	}
```

Comment:
I prefer this style to avoid unnecessary if check.

```suggestion
	if err != nil {
		return err
	}
	if res.GetHeader().GetError() != nil {
		return errors.Errorf("bootstrap failed: %s", resp.Header.GetError().String())
	}
```